### PR TITLE
Add non legacy builders (<10.9)

### PR DESCRIPTION
**Pull request by arjunsalyan**
I will soon be doing the table related changes, but till then it would be better to not miss on the build history for the new builders.

https://github.com/macports/macports-webapp/issues/117

### CODE REVIEW

**Comment by arjunsalyan** [comment]
I have applied this to the test site:

http://ec2-52-34-234-111.us-west-2.compute.amazonaws.com/ports/all_builds/

**Comment by mojca** [comment]
I would ideally like to see just one instance of 10.6 on port health status, and just one entry in the filter.

**Comment by arjunsalyan** [comment]
> I would ideally like to see just one instance of 10.6 on port health status, and just one entry in the filter.

I am working on the new table design, shall I simply hardcode the `os_arch_key` and `display_name` columns?

**Comment by arjunsalyan** [comment]
This PR is not in the stage of merging right now, as it will cause migration conflicts with https://github.com/macports/macports-webapp/pull/126 .

**Comment by arjunsalyan** [comment]
I have deployed this on the test site. I am utilising only one additional field `display_name`.

Does it seem right now? @mojca 
http://ec2-52-34-234-111.us-west-2.compute.amazonaws.com

**Comment by mojca** [comment]
Can you please try to import at least a few builds (even if just ten builds per worker)? I don't yet see any builds shown after the 22nd of August to be able to see how it works.

**Comment by arjunsalyan** [comment]
> Can you please try to import at least a few builds (even if just ten builds per worker)? I don't yet see any builds shown after the 22nd of August to be able to see how it works.

I have started the process, the fetching should finish within next 10-15 minutes.

**Comment by arjunsalyan** [comment]
Also, please do a hard refresh in the browser so that the changes to the javascript get reflected.

**Comment by mojca** [comment]
Can you please point me to some actual builds?

**Comment by arjunsalyan** [comment]
> Can you please point me to some actual builds?

On the page: http://ec2-52-34-234-111.us-west-2.compute.amazonaws.com/ports/all_builds/ select builder `10.8`.
Now, the first 20 builds refer to the builder `10.8_x86_64` (2285 to 2304) while the builds below it refer to the builder `10.8_x86_64_legacy`.

The app is initially set to fetch only 20 builds for those builders which do not have any builds in the database, that is why we only get the latest 20 builds for the newly added `10.8_x86_64`. We will need to decide this number.

And I also missed adding non-legacy builder for `10.6_i386`.

**Comment by mojca** [comment]
Perfect, thank you. They are there now, but I didn't see them earlier.

The all_builds works precisely as desired. The port health on the other hand is picking up the legacy workers, maybe because you sorted by build number (but I didn't check).

That said, the port health has other flaws, like: the user doesn't see if the worker is offline, or the latest build is still pending (10.5/ppc has been offline for ages and port health shows just whatever was built last); I would attempt to fix this with buildbot 2.

Fetching just the latest 20 builds sounds fine. But we would need to distinguish between two different scenarios:
* users or temporary servers who just want to test a new feature (we don't want them to fetch all builds, at least not automatically)
* production server where we really want to fetch all builds (and we need to try to do this before they go "out of focus" and get deleted)

I'm ok with merging this now and fixing the issue with port health immediately after that.

File: app/ports/views.py [hunk 1]
```diff
@@ -87,15 +87,15 @@ def portdetail(request, name, slug="summary"):
         if tab not in allowed_tabs:
             return HttpResponse("Invalid tab requested. Expected values: {}".format(allowed_tabs))
 
-        all_latest_builds = BuildHistory.objects.all().order_by('port_name', 'builder_name', '-build_id').distinct('port_name', 'builder_name')
-        port_latest_builds = list(BuildHistory.objects.filter(id__in=Subquery(all_latest_builds.values('id')), port_name__iexact=name).values('builder_name__name', 'build_id', 'status'))
+        all_latest_builds = BuildHistory.objects.all().order_by('port_name', 'builder_name__display_name', '-build_id').distinct('port_name', 'builder_name__display_name')
+        port_latest_builds = list(BuildHistory.objects.filter(id__in=Subquery(all_latest_builds.values('id')), port_name__iexact=name).values('builder_name__name', 'builder_name__display_name', 'build_id', 'status'))
```
Comment (mojca) [inline app/ports/views.py]: We can no longer sort by build id. Maybe by `time_start`?

File: app/ports/migrations/0011_populate_display_name.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# Generated by Django 2.2.4 on 2019-10-09 14:19
+
+from django.db import migrations
+
+
+def add_new_builders(apps, schema_editor):
+    Builder = apps.get_model('ports', 'Builder')
+    Builder.objects.create(name='10.8_x86_64')
+    Builder.objects.create(name='10.7_x86_64')
+    Builder.objects.create(name='10.6_x86_64')
```
Comment (mojca) [inline app/ports/migrations/0011_populate_display_name.py:10]: As already noted, we are missing the 10.6_i386 one.

**Review by mojca — APPROVED** [review]
Two hopefully minor changes to fix, and then we can merge it.

And then separately think of the best way to fetch all builds just on production machine.

File: app/ports/views.py (hunk 1 above)
Comment (arjunsalyan) [reply app/ports/views.py]: Yes, thank you.

**Comment by arjunsalyan** [comment]
This is probably ready to be merged now? I am sorry for the delay.

**Comment by mojca** [comment]
Cool, perfect.

Now we just need a way to fetch the old builds (fetch them all on the main server without having to fetch them on some test instances).

**Comment by mojca** [comment]
Ok, we are still missing a few fixes. Looking at http://ec2-52-34-234-111.us-west-2.compute.amazonaws.com/port/qt59-qtserialport/, the builds are still picking the wrong builder. It works correctly if you switch to a different tab.

It might make sense to clean up the code a bit, but for now at least we should fix that bug.

**Comment by arjunsalyan** [comment]
> Ok, we are still missing a few fixes. Looking at http://ec2-52-34-234-111.us-west-2.compute.amazonaws.com/port/qt59-qtserialport/, the builds are still picking the wrong builder. It works correctly if you switch to a different tab.

I am sorry, could you please specify it more? Also, the recent changes (made today) have not been deployed to the test site due to migration conflicts.

**Comment by mojca** [comment]
> I am sorry, could you please specify it more?

Hmmm ... I can no longer reproduce what I saw. The most likely explanation is that I was accidentally looking/clicking at the 10.7 builder which still points to the legacy build. It might have to do with me travelling on a bumpy road while trying to concentrate on the screen :) :) :)

**Pull request merged**